File: python/common/support_ticket.py
"""
Support Ticket Models

Represents a customer support ticket and related enums.
"""

from dataclasses import dataclass
from enum import Enum


class TicketPriority(Enum):
    """Ticket priority levels."""
    LOW = "Low"
    MEDIUM = "Medium"
    HIGH = "High"
    CRITICAL = "Critical"


@dataclass
class SupportTicket:
    """Represents a customer support ticket."""
    ticket_id: str
    customer_id: str
    customer_name: str
    subject: str
    description: str
    priority: TicketPriority = TicketPriority.MEDIUM

File: dotnet/Common/SupportTicket.cs
namespace WorkflowLab.Common;

/// <summary>
/// Represents a customer support ticket.
/// </summary>
public sealed record SupportTicket(
    string TicketId,
    string CustomerId,
    string CustomerName,
    string Subject,
    string Description,
    TicketPriority Priority = TicketPriority.Medium
);

/// <summary>
/// Ticket priority levels.
/// </summary>
public enum TicketPriority
{
    Low,
    Medium,
    High,
    Critical
}
